internal/core/docker_context_test.go

```diff
@@ -0,0 +1,55 @@
+package core
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// setupDockerContexts creates a temporary directory structure for testing the Docker context functions.
+// It creates the following structure, where $i is the index of the context, starting from 1:
+// - $HOME/.docker
+//   - config.json
+//   - contexts
+//   - meta
+//   - context$i
+//   - meta.json
+//
+// The config.json file contains the current context, and the meta.json files contain the metadata for each context.
+// It generates the specified number of contexts, setting the current context to the one specified by currentContextIndex.
+func setupDockerContexts(t *testing.T, currentContextIndex int, contextsCount int) {
```

Comment:
question: this looks like it could trash the user configuration?

Reply:
It's used in tests, and it's always used in conjunction with t.SetEnv("HOME"). But you're totally right that, in general, not only here, there are unit tests that need to be refactored and leverage tmp dirs more and more to avoid exactly that: messing up with the current user conf.

Reply:
Definitely in favor of avoiding interacting with the actual home dir, if possible.

Reply:
I'd work on that refinement in a separate PR, to not mix things up.